dart/lib/src/sentry_client.dart

```diff
@@ -555,7 +555,38 @@ class SentryClient {
         }
       }
     }
+
     if (processedLog != null) {
+      final contexts = Contexts();
+      for (final enricher in _options.contextsEnrichers) {
+        await enricher.enrich(contexts);
+      }
```

Comment:
I'm still thinking about the `on/emit` method.

Can't we register a hook in Flutter init where within we enrich the log? that way we don't need to pass contexts enricher between dart and flutter

e.g

```dart
// in flutter package
client.on('beforeCaptureLog', (log) {
  // get os and device info and add them to the log attributes
});
```